boshio/stemcells.go

```diff
@@ -36,18 +37,14 @@ func (s Stemcells) FindStemcellByVersion(version string) (Stemcell, bool) {
 }
 
 func (s Stemcells) FilterByType() Stemcells {
-
-	if s.supportsLight() == false {
-		// all stemcells are Regular, no need to filter
-		return s
-	}
-
 	if s.forceRegular() {
 		return s.regularStemcellsOnly()
+	} else if s.forceLight()  {
+		return s.lightStemcellsOnly()
+	} else if s.supportsLight() == false {
+		// all stemcells are Regular, no need to filter
+		return s
 	} else {
-		// The light stemcells might be published several hours after the regular versions
-		// The resource should wait until the corresponding light version is available to avoid
-		// caching a bulky regular stemcell
 		return s.lightStemcellsOnly()
```

Comment:
in the description of `force_light` you mention
```
Without `force_light` set it is possible for a `check` to return a regular stemcell in
  cases where the regular has been published, but the light version is not yet published.
```

this is against the code here. I wonder is `s.lightStemcellsOnly()` a wrong name since it could return regular? Or was the previous comment that got deleted above not accurate?

Reply:
`s.lightStemcellsOnly()` is correctly named, that function will never return a regular stemcell.

The problem previously was the `if s.supportsLight() == false {` conditional wasn't always correct.

`supportsLight` assumes that if it sees a single stemcell in the list, that light stemcells are supported. This is generally true, but falls apart when the list of stemcells is just a list of one because it's an incremental check.

The alternative way to fix this would be to determine if the stemcell `supportsLight` before the list of stemcells is filtered. But I believe currently the main filtering is happening via the bosh.io api, so this would involving pulling down the full list of stemcells from bosh.io for every check. ~~Which is probably not an idea usage of resources for the concourse workers or the bosh.io servers.~~

Actually, I was wrong, I thought it used the previous check version to filter the api call to bosh.io, but it doesn't at all, it actually does fetch all the stemcells and filters them after the fact.

I'm going to close this PR and create a new one that just fixes `supportsLight` so it is correct. This should just fix the edge case behavior, without having to change the configuration interface at all.